MalwareAnalyzer/views.py

```diff
@@ -31,84 +31,33 @@ def apkid_analysis(app_dir, apk_file):
-            apkid_dict["obfuscator"] = apkid_dict["result"]["obfuscator"]
-        else:
-            apkid_dict["obfuscator"] = ""
-
-        if "abnormal" in apkid_dict["result"]:
-            apkid_dict["abnormal"] = apkid_dict["result"]["abnormal"]
-        else:
-            apkid_dict["abnormal"] = ""
-
-        if "anti_disassembly" in apkid_dict["result"]:
-            apkid_dict["anti_disassembly"] = apkid_dict[
-                "result"]["anti_disassembly"]
-        else:
-            apkid_dict["anti_disassembly"] = ""
-
-        if "dropper" in apkid_dict["result"]:
-            apkid_dict["dropper"] = apkid_dict["result"]["dropper"]
-        else:
-            apkid_dict["dropper"] = ""
-
-        if "manipulator" in apkid_dict["result"]:
-            apkid_dict["manipulator"] = apkid_dict["result"]["manipulator"]
-        else:
-            apkid_dict["manipulator"] = ""
-
-        apkid_dict["result"] = ""
-        if "files" in result:
-            apkid_dict["result"] = result["files"][1]["results"]
-        
-        if "anti_vm" in apkid_dict["result"]:
-            apkid_dict["anti_vm"] = apkid_dict["result"]["anti_vm"]
-        
-        if "compiler" in apkid_dict["result"]:
-            apkid_dict["compiler"] = apkid_dict["result"]["compiler"]
+        apkid_dict['anti_vm'] = apkid_dict['result'].get('anti_vm', '')    
+        apkid_dict['compiler'] = apkid_dict['result'].get('compiler', '')
+        apkid_dict['packer'] = apkid_dict['result'].get('packer', '')
+        apkid_dict['obfuscator'] = apkid_dict['result'].get('obfuscator', '')
+        apkid_dict['abnormal'] = apkid_dict['result'].get('abnormal', '')
+        apkid_dict['anti_disassembly'] = apkid_dict[
+                "result"].get('anti_disassembly', '')
+        apkid_dict['anti_debug'] = apkid_dict['result'].get('anti_debug', '')
+        apkid_dict['dropper'] = apkid_dict['result'].get('dropper','')
+        apkid_dict['manipulator'] = apkid_dict['result'].get('manipulator', '')
+        apkid_dict['result'] = ''
         
-        if "packer" in apkid_dict["result"]:
-            apkid_dict["packer"] = apkid_dict["result"]["packer"]
-        
-        if "obfuscator" in apkid_dict["result"]:
-            apkid_dict["obfuscator"] = apkid_dict["result"]["obfuscator"]
-        
-        if "abnormal" in apkid_dict["result"]:
-            apkid_dict["abnormal"] = apkid_dict["result"]["abnormal"]
-        
-        if "anti_disassembly" in apkid_dict["result"]:
-            apkid_dict["anti_disassembly"] = apkid_dict[
-                "result"]["anti_disassembly"]
-        
-        if "dropper" in apkid_dict["result"]:
-            apkid_dict["dropper"] = apkid_dict["result"]["dropper"]
-        
-        if "manipulator" in apkid_dict["result"]:
-            apkid_dict["manipulator"] = apkid_dict["result"]["manipulator"]
+        if "files" in result:
+            apkid_dict2["result"] = result["files"][1]["results"]

```

Comment:
1 Will not get the value?

Reply:
I see rare case without value for position 1